include/picongpu/param/binningSetup.param

```diff
@@ -0,0 +1,216 @@
+
+
+            if constexpr(NameOrTypeExists<decltype(electronsObj)>::value)
+            {
+                // Bring the species together in a tuple
+                auto speciesTuple = createTuple(electronsObj);
+
+                /**
+                 * Define deposited quantity here
+                 */
+                auto getParticleChargeDensity = [] ALPAKA_FN_ACC(auto const& worker, auto const& particle) -> float_X
+                {
+                    const float_X charge = attribute::getCharge(particle[weighting_], particle);
+                    const float_X particleChargeDensity = precisionCast<float_X>(charge / CELL_VOLUME);
+                    return particleChargeDensity;
+                };
+
+                std::array<double, 7> depositedUnits{}; // Tell user the 7 dimensional format
+                depositedUnits[SIBaseUnits::length] = -3.0;
+                depositedUnits[SIBaseUnits::electricCurrent] = 1.0;
+                depositedUnits[SIBaseUnits::time] = 1.0;
+
+                // @todo enforce functor return type is same as createFunctorDescription template type
+                auto chargeDepositionData
+                    = createFunctorDescription<float_X>(getParticleChargeDensity, "ChargeDensity", depositedUnits);
+
+
+                auto getParticleCount
+                    = [] ALPAKA_FN_ACC(auto const& worker, auto const& particle) -> float_X { return 1.0; };
+                auto getCounts = createFunctorDescription<float_X>(getParticleCount, "Counts");
+
+                /**
+                 * define notify period, dump frequency, normalizeByBinVolume, time averaging
+                 * NNote that steps in time averaging are "notify steps" not simulation steps
+                 */
+                uint32_t dataDumpNNotifies = 0u;
+                bool enableTimeAveraging = true;
+                std::string notifyPeriod = "1";
+                bool normalizeByBinVolume = true;
+
+                /**
+                 * Gives access to series, iteration and mesh to the user
+                 * the iteration is already initialized as ```series.writeIterations()[currentStep];```
+                 * Please dont use the series to do funky stuff with the iteration, unless you know what you are
+                 * doing
+                 * This is called before any other mesh data is written to openPMD
+                 * Make sure to capture by copy only, as the objects in this file are not kept alive
+                 */
+                auto writeOpenPMD =
+                    [ax_py](::openPMD::Series& series, ::openPMD::Iteration& iteration, ::openPMD::Mesh& mesh) -> void
+                {
+                    mesh.setAttribute("p_nbins", ax_py.getNBins());
+                    // mesh.setAttribute("p_min", ax_py.getMin()); // This line throws an error in particle
+                    // calorimeter mesh.setAttribute("p_max", ax_py.getMax()); // This line throws an error in
```

Comment:
Can you try splitting this on multiple lines? I'd like to see if the error comes from `getMin()` or from `setAttribute()`. Given that it's an error somewhere inside CUDA, it's most likely the `getMin()` and `getMax()` thing. I can imagine that copying `ax_py` into the closure might at this point not be enough to keep the underlying data alive/valid.

Reply:
The copy into the closure works, and writes output as expected. 
I tried recreating the error, and though I was struggling with it for a long time, I can no longer trigger it. 
I first thought maybe the error went away when I renamed the `binningSetup` from `.hpp` to `.param`, but that's not the case. So, I will for now ignore this until it breaks in the future. I also cleaned up some includes and namespaces, so that might also have helped at some point.  
Sorry about the hunt that went nowhere, but thanks for being the first person to pull and compile the plugin :smile:, I've updated the documentation with the things learnt from your first use.